File: src/Poker/hand/HighCard.java
package Poker.hand;

import Poker.game.DeckConstants.Value;

public class HighCard extends Hand
{
   private Value cardValues[];
   
   public HighCard(Value values[])
   {
      super(Rank.HIGH_CARD);
      cardValues = new Value[5];
      for (int i = 0; i < 5; i++)
      {
         cardValues[i] = values[i];
      }
   }
   
   public Value getCardValue(int i)
   {
      return cardValues[i];
   }
   
   public String toString()
   {
      return cardValues[0].toString() + " high";
   }
   
   public int compare(Hand handToCompare)
   {
      int iCompare = compareRank(handToCompare);
      if (iCompare != 0)
      {
         return iCompare;
      }
      
      return compareHighCards((HighCard) handToCompare);
   }
   
   private int compareHighCards(HighCard hc)
   {
      for (int i = 0; i < 5; i++)
      {
         if (cardValues[i] != hc.getCardValue(i))
         {
            return cardValues[i].getIndex() - hc.getCardValue(i).getIndex();
         }
      }
      
      return 0;
   }
}

File: server/hand/highCard.js
let Hand = require('./hand');

let HighCard = (values) =>
{
	const HIGH_CARD = 0;
	
	let rank = Hand(HIGH_CARD);
	
	let self =
	{
		getRank: () => rank,
		compare: (hand) =>
		{
			let rankCompare = rank.compare(hand.getRank());
			if (rankCompare)
			{
				return rankCompare;
			}
			
			return self.compareSameRank(hand);
		},
		compareSameRank: (highCard) =>
		{
			for (let i = 0; i < 5; i++)
			{
				if (values[i] !== highCard.getValue(i))
				{
					return values[i] - highCard.getValue(i);
				}
			}
			
			return 0;
		},
		getValue: (i) => values[i],
		toString: () =>
		{
			let valuesStr = '';
			for (let i = 0; i < 4; i++)
			{
				valuesStr += (values[i] + ', ');
			}
			valuesStr += values[4];
			
			return 'High card (' + valuesStr + ')';
		}
	}
	
	return self;
}

module.exports = HighCard;
